### DISCUSSION TITLE
trying to make socketio work with azure web app

### DISCUSSION THREAD

**Discussion by ldariva** [miguelgrinberg/Flask-SocketIO#1923] [Q&A]
Hi @miguelgrinberg ,

First of all I would like to say that I am a big fan of your work. It has saved my life hundreds of time. 

I am developing a web app that among other thing receive an excel file and process it line by line. I created a progress bar which is updated using socket-io communication. In the development environment it works (and it was pretty easy to make it work). However when I upload my code to azure I am not being able to connect to the socket-io. I followed this example [https://github.com/vjammar/aws-iot-python](https://github.com/vjammar/aws-iot-python). As I am still in developing it I am using a free linux app (minimum resources) in azure. According to this [page](https://learn.microsoft.com/en-us/troubleshoot/azure/app-service/faqs-app-service-linux#web-sockets) from MS the websocket is always enable now (some time ago, it was necessary to manually enable it)

This is how I configured the socket io on the flask app in __init__.py

```
from flask import Flask
...
from flask_socketio import SocketIO,emit
...
socketio = SocketIO(async_mode=None, logger=True, engineio_logger=True)

def create_app(config_name):

    app = Flask(__name__)
    app.config.from_object(config[config_name])
    config[config_name].init_app(app)
    
    with app.app_context():
        ...
        socketio.init_app(app)
        
    from .main import main as main_blueprint
    app.register_blueprint(main_blueprint)
    
    return app
```

In the client side I am trying to connet to the socket io like this. Instances is 

```
var socket = io.connect('https://<myapp>.azurewebsites.net:433/instances/',{
'reconnect': true,
'reconnection delay': 500,
'max reconnection attempts': 10
 });
```

When I try to submit the file I receive the following error

socket.io.js:3941 
       GET https://ampsystem.azurewebsites.net:433/socket.io/?EIO=4&transport=polling&t=OJsy8J6 net::ERR_CONNECTION_TIMED_OUT

Could you give me some hint?
Thanks

**Comment by miguelgrinberg** [comment]
Please do not use the issues board for questions. There is this discussions board for that. Thank you.

Unfortunately I'm not familiar with Azure, so I'm not sure how I can help with this. Is the server receiving the WebSocket request at all? If it isn't then the problem has to be in the configuration of your Azure instance.

**Reply by ldariva** [reply]
Hi Miguel.
Sorry for that. 
Do you know how can I test if the server is receiveing the request?
Thanks

**Reply by miguelgrinberg** [reply]
Does your server generate logs? That would be the normal way to check what is received.

**Reply by ldariva** [reply]
I am receiving this message in azure log
The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details. (further occurrences of this error will be logged with level INFO)
Any thoughts?
Thks

**Reply by miguelgrinberg** [reply]
@ldariva the error tells you what needs to be done. Is it not clear?

**Reply by ldariva** [reply]
No. It does not say anything else. I submited the issue tô MS support. Let's see what they say. Thanks

**Reply by miguelgrinberg** [reply]
This is an error message from this package, not from Azure. You have to do what the error message says:
- Install a WebSocket server that is compatible with the async mode you have chosen to use
- See the documentation if you need help doing the above
